.github/scripts/Tests/session-cleanup-detector.Tests.ps1

```
@@ -1517,3 +1517,183 @@ Describe 'Get-OrphanBranchLines — per-line suffix and composite invocation' {
+        $unregistered = @($writtenBasenames | Where-Object { $_ } | Where-Object { $registeredFilenames -notcontains $_.ToLowerInvariant() })
+        $unregistered | Should -BeNullOrEmpty `
+            -Because "oracle-found root-level tracking writers must all be enrolled in Get-SCDPersistentTrackingExclusions: missing: $($unregistered -join ', ')"
+    }
+
+    It 'AC7: no persistent-filename literal exists outside Get-SCDPersistentTrackingExclusions' {
+        $repoRoot = (Resolve-Path (Join-Path $PSScriptRoot '../../..')).Path
+
+        $exclusions = Get-SCDPersistentTrackingExclusions
+
+        foreach ($fname in $exclusions.Filenames) {
+            $matches = @(
```

PSScriptAnalyzer (1.25.0)

[warning] The Variable 'matches' is an automatic variable that is built into PowerShell, assigning to it might have undesired side effects. If assignment is not by design, please use a different name.

(PSAvoidAssignmentToAutomaticVariable)